src/Features/Core/Portable/PdbSourceDocument/PdbSourceDocumentMetadataAsSourceFileProvider.cs

```diff
@@ -151,7 +162,7 @@ public PdbSourceDocumentMetadataAsSourceFileProvider(IPdbFileLocatorService pdbF
                     }
                 }
 
-                using (var textWriter = new StreamWriter(tempFilePath, append: false, encoding: MetadataAsSourceGeneratedFileInfo.Encoding))
+                using (var textWriter = new StreamWriter(tempFilePath, append: false, encoding: text.Encoding ?? Encoding.UTF8))
```

Comment:
So I'm a bit confused here -- we're getting text loaders that decoded the stream into text, and then we're encoding this back into a stream again. Why just not copy the underlying stream directly?

Reply:
(more generally, are we extracting all the source out of the PDB in one go? If so, should we just be making FileTextLoaders that directly point to the files on disk? Otherwise we're pulling in all the text here and then rooting it for the process lifetime, right?)

Reply:
Yeah, this will change in a follow up. For now this supports embedded text, which needs to be written to a temp file here, and files on disk which don't need to be written, but also are probably extremely rare. The next step is to integrate the debugger bits which will be giving us a file path from their symbol cache, which we absolutely have to use directly, so this bit of code will move to just apply to embedded source.